traveltimepy/mapper.py

```diff
@@ -0,0 +1,510 @@
+import math
+from datetime import datetime
+from typing import Dict, Union, List, Optional
+
+from traveltimepy.errors import ApiError
+from traveltimepy import TimeFilterFastRequest_pb2
+
+from traveltimepy.dto.common import Location, LocationId, Coordinates, FullRange, Property, Range
+from traveltimepy.dto.transportation import PublicTransport, Driving, Ferry, Walking, Cycling, DrivingTrain
+from traveltimepy.dto.requests.time_filter_fast import Transportation
+from traveltimepy.dto.requests.zones import ZonesProperty
+
+from traveltimepy.dto.requests.postcodes import PostcodesRequest
+from traveltimepy.dto.requests.routes import RoutesRequest
+from traveltimepy.dto.requests.time_filter import TimeFilterRequest
+from traveltimepy.dto.requests.time_filter_fast import TimeFilterFastRequest
+from traveltimepy.dto.requests.time_filter_proto import ProtoTransportation
+from traveltimepy.dto.requests.zones import DistrictsRequest, SectorsRequest
+from traveltimepy.dto.requests.time_map import TimeMapRequest
+
+
+from traveltimepy.dto.requests import (
+    time_filter,
+    time_filter_fast,
+    postcodes,
+    zones,
+    routes,
+    time_map
+)
+
+
+def create_time_filter(
+    locations: List[Location],
+    searches: Dict[LocationId, List[LocationId]],
+    transportation: Union[PublicTransport, Driving, Ferry, Walking, Cycling, DrivingTrain],
+    properties: Optional[List[Property]],
+    departure_time: Optional[datetime],
+    arrival_time: Optional[datetime],
+    travel_time: int,
+    full_range: Optional[FullRange]
+) -> TimeFilterRequest:
+    if properties is None:
+        properties = [Property.TRAVEL_TIME]
+    if arrival_time is not None:
```

Comment:
If both `arrival_time` and `departure_time` is set then we silently do an arrival search.

Reply:
yeah, good point, I'll add throwing errors there